mycore-base/src/main/java/org/mycore/common/xml/MCRXMLFunctions.java

```diff
@@ -683,6 +683,29 @@ public static String getDisplayName(String classificationId, String categoryId)
         }
     }
 
+    /**
+     * Get the display name for the given category in the desired language.
+     *
+     * @param classificationId the id of the classification
+     * @param categoryId the category id
+     * @param lang the desired language for which to get the display name
+     *
+     * @return the display name or an empty string when there is no label in the desired language
+     */
+    public static String getDisplayName(String classificationId, String categoryId, String lang) {
+        try {
+            MCRCategoryID categID = new MCRCategoryID(classificationId, categoryId);
+            MCRCategoryDAO dao = MCRCategoryDAOFactory.getInstance();
+            MCRCategory category = dao.getCategory(categID, 0);
+            Optional<MCRLabel> label = category.getLabel(lang);
+            return label.isEmpty() ? "" : label.get().getText();
```

Comment:
I don't know the use case for this method no there is no further information given in the ticket description.

Wouldn't it be better to fall back to `category.getCurrentLabel()` if there is no label in the desired language, rather then an empty string?

Reply:
I just want an easy method for getting a label in a certain language with basically the same behaviour as `org.mycore.common.xml.MCRXMLFunctions#getDisplayName(java.lang.String, java.lang.String)`. That is the reason an empty string is returned if a label cannot be resolved.